lib/tpm_tcg2.c

```diff
@@ -468,8 +468,19 @@ static int tcg2_log_parse(struct udevice *dev, struct tcg2_event_log *elog)
 	}
 
```

Comment:
There's another check a few lines above in 
```
for (i = 0; i < count; ++i) {
                algo = get_unaligned_le16(&event->digest_sizes[i].algorithm_id);
                mask = tcg2_algorithm_to_mask(algo);
        
                if (!(active & mask))
                        return 0;
```
which you ignore, that's a problem as well